MaxText/layers/attentions.py

```diff
@@ -273,6 +273,8 @@ class AttentionOp(nn.Module):
   use_ragged_attention: bool = False
   ragged_block_size: int = 256
 
+  scale_factor: float = 1  # scaling factor for query in attention. currently used in cudnn only.
```

Comment:
let's not have an extra config unless there is a strong reason for that, we already have a way to scale query, e.g. `query_pre_attn_scalar` which has the same effect on the attention.

Reply:
reused the `query_pre_attn_scalar` for the scaling